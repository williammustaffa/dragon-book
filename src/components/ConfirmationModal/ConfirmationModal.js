import React from "react"
import PropTypes from "prop-types";
import { Button, Modal } from "semantic-ui-react"

function ConfirmationModal({ open, title, text, onConfirm, onClose }) {
  function onConfirmClick() {
    if (typeof onConfirm === "function") {
      onConfirm();
    }
  }

  return (
    <Modal size="tiny" open={open} onClose={onClose}>
      <Modal.Header>{title}</Modal.Header>
      <Modal.Content>
        <p>{text}</p>
      </Modal.Content>
      <Modal.Actions>
        <Button
          color="red"
          onClick={onClose}
          icon="close"
          labelPosition="right"
          content="No"
        />
        <Button
          secondary
          onClick={onConfirmClick}
          icon="checkmark"
          labelPosition="right"
          content="Yes"
        />
      </Modal.Actions>
    </Modal>
  )
}

ConfirmationModal.propTypes = {
  open: PropTypes.bool,
  title: PropTypes.string,
  text: PropTypes.string,
  onConfirm: PropTypes.func,
  onClose: PropTypes.func,
};

export default ConfirmationModal;