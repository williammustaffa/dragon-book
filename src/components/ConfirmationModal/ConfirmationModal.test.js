import React from "react";
import ConfirmationModal from "./ConfirmationModal";
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

test("Test ConfirmationModal component", () => {
  const { asFragment } = render(<ConfirmationModal />);

  // Run tests
  expect(asFragment()).toMatchSnapshot();
});