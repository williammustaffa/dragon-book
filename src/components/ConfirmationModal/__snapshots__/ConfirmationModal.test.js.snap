// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ConfirmationModal component 1`] = `<DocumentFragment />`;
